internal/v1/handler_compose_image.go

```diff
@@ -467,6 +467,85 @@ func (h *Handlers) buildCustomRepositories(ctx echo.Context, custRepos []CustomR
 	return res, nil
 }
 
+func (h *Handlers) buildTemplateRepositories(ctx echo.Context, templateID string) ([]composer.Repository, []composer.CustomRepository, error) {
```

Comment:
This is what I had in mind indeed, and we should call either buildSnapshotRepositories or buildTemplateRepositories.  There might be a lot of overlap between the two however, so depending on what that looks like maybe we can refactor a little.